tajo-catalog/tajo-catalog-server/src/main/java/org/apache/tajo/catalog/CatalogServer.java

```diff
@@ -1195,6 +1249,71 @@ public GetTablePartitionsResponse getAllPartitions(RpcController controller, Nul
     }
 
     @Override
+    public GetTablePartitionsResponse getPartitionsByDirectSql(RpcController controller,
```

Comment:
Catalog API is public APIs. In terms of this point, this API design does not make sense due to the following problems:
 * Users should know its schema and underlying persistent storage.
  * So, users should make SQL statements for all persistent storages.
 * This approach is very prone to SQL injection.